Tests/SwiftDocCTests/XCTestCase+LoadingTestData.swift

```diff
@@ -138,11 +138,14 @@ extension XCTestCase {
         var analyzer = SemanticAnalyzer(source: nil, context: context, bundle: bundle)
         let result = analyzer.visit(blockDirectiveContainer)
         
-        let problemIDs = analyzer.problems.map { problem -> String in
-            let line = problem.diagnostic.range?.lowerBound.line.description ?? "unknown-line"
+        let problemIDs = try analyzer.problems.map { problem -> (line: Int, id: String) in
+            let line = try XCTUnwrap(problem.diagnostic.range).lowerBound.line
             
-            return "\(line): \(problem.diagnostic.severity) – \(problem.diagnostic.identifier)"
-        }.sorted()
+            return (line, "\(line): \(problem.diagnostic.severity) – \(problem.diagnostic.identifier)")
+        }
+        .sorted(by: \.id)
```

Comment:
The double sort feels really confusing. there must be a better way of achieving this. If we just need them sorted by line number then let's just keep the second sort. If we then need ordering on other criteria let's make it explicit and do a simple, `.sorted`.

Reply:
Ah I'm open to suggestions. We need it sorted on the line number so that the tests are readable but then we need the sort to be stable between test runs for multiple diagnostics that appear on the same line.

I considered dropping into a full `sorted` but I think those are inherently much more confusing to read.

Reply:
Oh I just hadn't considered sorting the tuple itself – will that work? Agreed that's a much nicer solution if so.

Reply:
Hm yeah a (Int, String) isn't comparable so just `sorted()` doesn't work unfortunately. I'll implement it manually and see how it is.

Reply:
Okay I moved to a custom `.sorted`. Let me know what you think.

Reply:
Interestingly enough you could write the sorted as `.sorted { $0 < $1 } because tuples somehow implement `<` without conforming to comparable apparently. I still think your version is better though